jina/drivers/__init__.py

```diff
@@ -336,18 +336,13 @@ def pre_traverse(_docs, recur_on, context_doc=None,
             _traverse(working_docs, 'chunks', None, 'granularity', self._granularity_start, self._granularity_end)
 
         if 'matches' in self._recur_on:
-
-            for d in docs:
```

Comment:
I think this change may be dangerous, the recursion was never equal.

Reply:
Hmm but what it really changes is: instead of going deep on every top-level document one-by-one, go deep by all together. In theory there should be no difference. During fixing of the unit-tests I'll see how it hurts.

Reply:
I just wanted to raise attention